processing/src/main/java/org/apache/druid/segment/transform/Transformer.java

```diff
@@ -170,15 +174,16 @@ public long getTimestampFromEpoch()
       }
     }
 
+    @Override
+    public long getTimestampFromEpoch()
+    {
+      return timestamp;
+    }
+
     @Override
     public DateTime getTimestamp()
     {
-      final RowFunction transform = transforms.get(ColumnHolder.TIME_COLUMN_NAME);
-      if (transform != null) {
-        return DateTimes.utc(getTimestampFromEpoch());
-      } else {
-        return row.getTimestamp();
-      }
+      return DateTimes.utc(getTimestampFromEpoch());
```

Comment:
Why not cache the `DateTime` object instead of the `long`? By doing that we can save many temp objects creation.